### PR TITLE
Progressively enhance folder permissions

### PR DESCRIPTION
**Pull request by tombye**
Gives admin users some extra UI so selecting folder permissions doesn't have to involve scanning all the folders.

https://www.pivotaltracker.com/story/show/165059135

folder_permissions

Can be seen on these pages:
- /services/<service_id>/users/invite
- /services/<service_id>/templates/folders/<folder_id>/manage
- /services/<service_id>/users/<user_id>

## Tests

It has unit tests for the JavaScript and has been tested on our most-used browsers:
- Safari on OSX
- Chrome on OSX
- Firefox on OSX
- IE11 on Win 7
- IE11 on Win 10
- Edge 15 on Win 10
- Safari 11 on iPhone
- Chrome on Win 7
- Chrome on Win 10
- Firefox on Win 10

### CODE REVIEW

**Comment by quis** [comment]
Tested locally, works great. Few tweaks I think we need (can talk through these tomorrow):
- the wording of the _nothing selected_ message (for folders and team members) isn’t quite there
- I have a bigger version of the folder icon I can give you
- can we hide the ‘Change’ button in the expanded view?

**Comment by quis** [comment]
Added the asset for the new icon here: f274759

**Comment by tombye** [comment]
I've pushed some changes for the list of things to be done.
https://github.com/alphagov/notifications-admin/pull/2956/commits/7328649537ad2d015d6e2475f080bd472038a8f1 refactors it so there's only one button and we instead swap the text when expanding/collapsing. The rest are either fixes for bugs in the module JS, some in other JS that Jest found and one for a few issues with the new .svg file.

I also did some refactoring to help the module JS read easier now its HTML is made up of summary, fieldset and footer components.

File: app/assets/stylesheets/components/checkboxes.scss [hunk 1]
```diff
@@ -1,3 +1,36 @@
...
+    padding: .526315em .789473em .263157em 0px;
+    margin-bottom: $gutter / 2;
+
+    &:focus {
+      outline: none;
+    }
+  }
+
+  .selection-summary__text--folders {
+    padding-left: 51px;
```
Comment (quis) [inline app/assets/stylesheets/components/checkboxes.scss]:
```suggestion
    padding: 10px 15px 5px 51px;
```

File: app/assets/stylesheets/components/checkboxes.scss [hunk 2]
```diff
@@ -1,3 +1,36 @@
+.selection-summary {
+
+  .selection-summary__text {
+    padding: .526315em .789473em .263157em 0px;
```
Comment (quis) [inline app/assets/stylesheets/components/checkboxes.scss]:
```suggestion
    padding: 5px 0 0 0;
```

File: app/assets/stylesheets/components/checkboxes.scss [hunk 3]
```diff
@@ -1,3 +1,36 @@
+.selection-summary {
+
+  .selection-summary__text {
```
Comment (quis) [inline app/assets/stylesheets/components/checkboxes.scss]: Using tabular numbers here will stop the word ‘of’ jumping horizontally as the count changes:

jump

```suggestion
  .selection-summary__text {
      @include core-19($tabular-numbers: true);
```

no-jump

**Pull request merged**